File: Exercicio3.java
public class Exercicio3 {
    private final String alfabeto = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
    public String krypto(String frase) {
        frase = frase.toUpperCase();
        String novaFrase = "";
        for (int i = 0; i < frase.length(); i++) {
            char letraFrase = frase.charAt(i);
            int letra = this.alfabeto.indexOf(letraFrase);
            if (letra == -1) {
                novaFrase += letraFrase;
                continue;
            }
            if (letra + 18 > this.alfabeto.length() - 1) {
                int diff = this.alfabeto.length() - letra;
                novaFrase += this.alfabeto.charAt(this.alfabeto.length() - 18 - diff);
            } else {
                novaFrase += this.alfabeto.charAt(letra + 18);
            }
        }
        return novaFrase;
    }
}

File: Exercicio3.py
alfabeto = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
def krypto(frase):
    frase = frase.upper()
    novaFrase = ""
    tamanhoAlfabeto = len(alfabeto)
    for i in range(len(frase)):
        letra = frase[i]
        try:
            posLetra = alfabeto.index(letra)
            if (posLetra + 18 > tamanhoAlfabeto - 1):
                diff = tamanhoAlfabeto - posLetra
                novaFrase += alfabeto[tamanhoAlfabeto - 18 - diff]
            else:
                novaFrase += alfabeto[posLetra + 18]
        except:
            novaFrase += letra
    return novaFrase
print(krypto("4S90S U64796C JN 3S9S51SA 5S XW09S"))
